app/src/main/java/com/movies/data/repository/SearchRepositoryImpl.kt

```diff
@@ -9,12 +10,16 @@ import com.movies.domain.repository.SearchRepository
 
 class SearchRepositoryImpl(
     private val fetchSearchedMovies: ServiceApi,
-    private val movieListDTOMapper: MovieListDTOMapper
-) : SearchRepository {
+    private val movieListDTOMapper: MovieListDTOMapper,
+    private val favouriteMoviesDao: FavouriteMoviesDao,
+    ) : SearchRepository {
     override suspend fun searchMovies(query: String): List<MovieDomainModel> {
         val remoteData = apiDataFetcher { fetchSearchedMovies.searchMovies(query) }
         if (remoteData is Resource.Success) {
-            return movieListDTOMapper(remoteData.data)
+            return movieListDTOMapper(remoteData.data).map {
+                it.isFavourite = favouriteMoviesDao.isFavouriteMovie(it.id)
```

Comment:
do you actually need this ? 
![image](https://github.com/DavitTsiskarashvili/Movies/assets/95241918/a703742b-0742-4a6a-9682-5db99d58f316)

Reply:
that is an optional search design. See the one on the left of that version, it has favourites too